### ISSUE TITLE
Add a filter for haploid/diploid in the phenotype annotation tables

### ISSUE THREAD

**Issue by kimrutherford** [pombase/website#1688]

**Comment by kimrutherford** [comment]
I've implemented this filter for the detailed view mode.  I'm still working on allowing filtering in the summary view as it's a bit trickier.

**Comment by kimrutherford** [comment]
>  I'm still working on allowing filtering in the summary view as it's a bit trickier.

That's done now.

Could you give it a test when you get a chance?

**Comment by ValWood** [comment]
- I only see the filter on the details view of query results, not on  summary view.

- However, a question. Does the filter need to be  present if the query was restricted to haploid? (it is currently, with diploid greyed out). This is OK<but I wanted to check that it is what we wanted (maybe we do, for consistency?).

- Also, the filter isn't currently appearing on the gene pages, only on query results pages.

**Comment by kimrutherford** [comment]
> not on summary view.

Sorry, I had forgotten to re-release the site.  The filter will be visible in the summary view on the gene, term and reference pages in 15 minutes or so.

> maybe we do, for consistency?

I thought it would be good for consistency, but it's easy to change.

> Also, the filter isn't currently appearing on the gene pages, only on query results pages.

I don't understand that.  Here it is on the page for sgo2: https://www.pombase.org/gene/SPAC15A10.15
This is the multi-locus "Cell phenotype" section in the detailed view:
![diploid-gene-page-1](https://user-images.githubusercontent.com/90474/116011846-26f96980-a67b-11eb-8c0c-dd136e6bf7b1.png)

**Comment by ValWood** [comment]
I don't know why I could not see it. It is there.

**Comment by kimrutherford** [comment]
I'll close this.  We can reopen if we see any problems.

**Comment by mah11** [comment]
I see a problem: The phenotype annotation filter for haploid/diploid isn't working for me. 

Symptom 1: Selecting any of the options in the "Haploid or Diploid" filter does nothing to the display.
Symptom 2: If either "Haploid" or "Diploid" is selected, the other filters don't work.
Symptom 3: If I change the ploidy filter back to"Haploid or Diploid", any other filters I've set immediately take effect and work thereafter (i.e. I can change any non-ploidy filter if, but only if, "Haploid or Diploid" is selected).

Tested just now using mre11, Firefox & Chrome on Linux; Firefox & Safari on Mac; symptoms same everywhere.

**Comment by kimrutherford** [comment]
Sorry about that.  Thanks for including the gene name as the bug wasn't occurring on the gene pages I've been using for testing.

It's fixed now but I've just spotted a bug in the summary view.  If I select "Diploid" on the mre11 page I end up with extensions shown without a genotype name:
![diploid-gene-page-bug-1](https://user-images.githubusercontent.com/90474/116070205-6f973e00-a6e0-11eb-8fa0-3e5912b2a2cc.png)

I've disabled the haploid/diploid in summary mode while I try to fix things.
